repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -118,46 +244,55 @@ def map_repositories(to_install):
                                 repo_not_mapped_pkgs)
 
 
-def process_events(events):
-    """ Process PES Events and generate Leapp messages """
-    to_install = {}
-    to_remove = {}
-
-    for event in events:
-        to_install.update(event.out_pkgs)
-
-        if event.action not in ('Present', 'Deprecated', 'Moved') and event.in_pkgs:
-            to_remove.update(event.in_pkgs)
+def report_skipped_packages(message, packages):
+    """Generate report message about skipped packages"""
+    title = 'Packages will not be installed'
+    summary = '{} {}\n{}'.format(len(packages), message, '\n'.join(['- ' + p for p in packages]))
+    reporting.report_generic(title=title, summary=summary, severity='high')
+    if is_verbose():
+        api.show_message(summary)
 
-    filter_by_repositories(to_install)
-    map_repositories(to_install)
 
+def filter_out_transaction_conf_pkgs(to_install, to_remove, transaction_configuration):
+    """
+    Filter out those PES events that conflict with the higher priority transaction configuration files.
+
+    :param to_install: A dict {<pkg_name>: <repository>} with pkgs to install based on PES data
+    :param to_remove: A dict {<pkg_name>: <repository>} with pkgs to remove based on PES data
+    :param transaction_configuration: RpmTransactionTasks model instance with pkgs to install, keep and remove based
+                                      on the user configuration files
+    """
+    # The two lists below are for debug message purposes only
+    pes_based_pkgs_not_to_be_installed = []
+    pes_based_pkgs_not_to_be_removed = []
+
+    for pes_based_pkg_to_install in to_install.keys():
+        if pes_based_pkg_to_install in transaction_configuration.to_remove:
+            to_install.pop(pes_based_pkg_to_install)
+            pes_based_pkgs_not_to_be_installed.append(pes_based_pkg_to_install)
+    for pes_based_pkg_to_remove in to_remove.keys():
+        if pes_based_pkg_to_remove in (transaction_configuration.to_install + transaction_configuration.to_keep):
+            to_remove.pop(pes_based_pkg_to_remove)
+            pes_based_pkgs_not_to_be_removed.append(pes_based_pkg_to_remove)
+
+    if pes_based_pkgs_not_to_be_installed:
+        api.current_logger().debug('The following packages will not be installed '
+                                   'because of the transaction configuration files in /etc/leapp/transaction/:'
```

Comment:
Please add whole path to the config. E.g. /etc/leapp/transaction/to_remove

Reply:
Done